src/Protocol/ForgeHandshake.h

```diff
@@ -0,0 +1,65 @@
+
+// ForgeHandshake.h
+
+// Implements Forge protocol handshaking
+
+#pragma once
+
+#include <stddef.h>
+
+// fwd:
+namespace Json
+{
+	class Value;
+}
+class cClientHandle;
+
+class cForgeHandshake
+{
+public:
+	cForgeHandshake(cClientHandle * client);
+
+	void AugmentServerListPing(Json::Value & ResponseValue);
+	void BeginForgeHandshake(const AString & a_Name, const AString & a_UUID, const Json::Value & a_Properties);
+
+	void SendServerHello();
+	void DataReceived(cClientHandle * a_Client, const char * a_Data, size_t a_Size);
+
+	AStringMap ParseModList(const char * a_Data, size_t a_Size);
```

Comment:
All public symbols should have doxy-comments about what they do, what parameter constaints they expect etc. For private symbols this is optional, but highly recommended as well.